service/src/main/java/snowplowservice/resources/SnowPlowResource.java

```diff
@@ -167,12 +173,33 @@ private ProcessedData filterData(List<LineString> lineStrings) {
         return processedData;
     }
 
+    private boolean getSnowInfo(List<Point> points) {
+        boolean currentlySnowing = false;
+
+        // should be 1 Point with weather details
+        Point weatherInfo = points.get(0);
```

Comment:
This will throw an `IndexOutOfBoundsException`. 

You need to call `isEmpty` before getting this item out of the List.